features/step_definitions/edit_restaurant.rb

```diff
@@ -0,0 +1,13 @@
+Given(/^the following restaurant exists$/) do |table|
+  table.hashes.each do |restaurant|
+    FactoryGirl.create(:restaurant, id: restaurant[:id],
```

Comment:
For every iteration `restaurant` is a hash. This means that you can probably do:
```
FactoryGirl.create(:restaurant, restaurant)
```
Try it. 
Also, since you are setting the `id` please use your debugger and inspect that element. Is the `id` you set in the table persistent?